### PR TITLE
[DOC] Share query editor doc for traces

### PR DESCRIPTION
**Pull request by knylander-grafana**
**What is this feature?**
This PR updates the query editor content for Tempo to reflect content written in the Tempo docs (see https://grafana.com/docs/tempo/latest/traceql/query-editor/).  

These changes are made to the docs: 
* Updates for content based upon Tempo docs
* Query editor has its own page
* Query builder has its own page
* Query builder and editor content are now in shared files

**Why do we need this feature?**
This PR helps make the content for query editor and query builder consistent across Grafana, Grafana Cloud, and Tempo. 

**Which issue(s) does this PR fix?**:

Fixes #https://github.com/grafana/tempo-squad/issues/305
Fixes #https://github.com/grafana/tempo-squad/issues/306

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [X] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: docs/sources/datasources/tempo/query-editor/index.md [hunk 1]
```diff
@@ -24,7 +24,43 @@ The Tempo data source's query editor helps you query and display traces from Tem
...
+To add TraceQL panels to your dashboard, refer to the [Traces panel documentation](/docs/grafana/latest/panels-visualizations/visualizations/traces/).
+
+To learn more about Grafana dashboards, refer to the [Use dashboards documentation](/docs/grafana/latest/dashboards/use-dashboards/).
+
+## Write TraceQL queries in Grafana
+
+Inspired by PromQL and LogQL, TraceQL is a query language designed for selecting traces.
+The default traces search reviews the whole trace.
+TraceQL provides a method for formulating precise queries so you can zoom in to the data you need.
+Query results are returned faster because the queries limit what is searched.
```
Comment (knylander-grafana) [inline docs/sources/datasources/tempo/query-editor/index.md:36]:
```suggestion
```

File: docs/sources/datasources/tempo/query-editor/index.md [hunk 2]
```diff
@@ -24,7 +24,43 @@ The Tempo data source's query editor helps you query and display traces from Tem
...
+
+To learn more about how to query by TraceQL, refer to the [TraceQL documentation](/docs/tempo/latest/traceql).
+
+You can compose TraceQL queries in Grafana and Grafana Cloud using **Explore** and a Tempo data source. You can use either the **Query type** > **Search** (the TraceQL query builder) or the **TraceQL** tab (the TraceQL query editor).
+Both of these methods let you build queries and drill-down into result sets.
```
Comment (knylander-grafana) [inline docs/sources/datasources/tempo/query-editor/index.md:41]:
```suggestion
You can compose TraceQL queries in Grafana and Grafana Cloud using **Explore** and a Tempo data source. You can use either the **Query type** > **Search** (the TraceQL query builder) or the **TraceQL** tab (the TraceQL query editor).
Both of these methods let you build queries and drill-down into result sets.

To learn more about how to query by TraceQL, refer to the [TraceQL documentation](/docs/tempo/latest/traceql).
```

File: docs/sources/datasources/tempo/query-editor/index.md [hunk 3]
```diff
@@ -24,7 +24,39 @@ The Tempo data source's query editor helps you query and display traces from Tem
...
 For general documentation on querying data sources in Grafana, see [Query and transform data][query-transform-data].
 
-## Query by search
+To add TraceQL panels to your dashboard, refer to the [Traces panel documentation](/docs/grafana/latest/panels-visualizations/visualizations/traces/).
+
+To learn more about Grafana dashboards, refer to the [Use dashboards documentation](/docs/grafana/latest/dashboards/use-dashboards/).
+
+## Write TraceQL queries in Grafana
+
+
```
Comment (knylander-grafana) [inline docs/sources/datasources/tempo/query-editor/index.md:33]:
```suggestion
```

**Review by lwandz13 — APPROVED** [review]
Looks great Kim!

**Pull request merged**